packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -252,6 +252,16 @@ function GeneratedInputField({
     )
   }
   if (fieldDefinition.type === DOCUMENT_UPLOADER_WITH_OPTION) {
+    ;(value as IFileValue[]).forEach((file) => {
```

Comment:
Use map here instead to create a new list. Mutating the value of `file.data` here, will reflect to all references of these files everywhere in the app. It's better to always create new values to avoid difficult to debug issues